exploration/text-vs-code.md

```diff
@@ -0,0 +1,216 @@
+users to format or change unimportant whitespace as much as they want.
+This avoids the need for translators or tools to be super pedantic about
+formatting.
+
+## Constraints
+
+Limiting the range of characters that need to be escaped in plain text is important.
+Following past precedent,
+this design doc will only consider encapsulation styles which
+start with `{` and end with `}`.
+
+The current syntax includes some plain-ascii keywords:
+`input`, `local`, `match`, and `when`.
+
+The current syntax and active proposals include some sigil + name combinations,
+such as `:number`, `$var`, `|literal|`, `+bold`, and `@attr`.
+
+The current syntax supports unquoted literal values as operands.
+
+Messages themselves are "simple strings" and must be considered to be a single
+line of text. In many containing formats, newlines will be represented as the local
+equivalent of `\n`.
+
+## Proposed Design
+
+### Start in text, encapsulate code, trim around statements
+
+Allow for message patterns to not be quoted.
+
+Encapsulate with `{…}` or otherwise distinguishing statements from
+the primarily unquoted translatable message contents.
+
+For messages with multiple variants,
+separate the variants using `when` statements.
+
+Trim whitespace between and around statements such as `input` and `when`,
+but do not otherwise trim any leading or trailing whitespace from a message.
+This allows for whitespace such as spaces and newlines to be used outside patterns
+to make a message more readable.
+
+Allow for a pattern to be `{{…}}` quoted
+such that it preserves its leading and/or trailing whitespace
+even when preceded or followed by statements.
+
+## Alternatives Considered
+
+### Start in code, encapsulate text
+
+This approach treats messages as something like a resource format for pattern values.
+Keywords are declared directly at the top level of a message,
+and patterns are always surrounded by `{{…}}` or some other delimiters.
+
+Whitespace in patterns is never trimmed.
+
+The `{{…}}` are required for all messages,
+including ones that only consist of text.
+Delimiters of the resource format are required in addition to this,
+so messages may appear wrapped as e.g. `"{{…}}"`.
+
+This option is not chosen due to adding an excessive
+quoting burden on all messages.
+
+### Start in text, encapsulate code, re-encapsulate text within code
+
+As in the proposed design, simple patterns are unquoted.
+Patterns in messages with statements, however,
+are required to always be surrounded by `{{…}}` or some other delimiters.
+
+This effectively means that some syntax will "enable" code mode for a message,
+and that patterns in such a message need delimiters.
+
+This option is not chosen due to adding an excessive
+quoting burden on all multi-variant messages,
+as well as introducing an unnecessary additional conceptual layer to the syntax.
+
+### Start in text, encapsulate code, trim minimally
```

Comment:
We need to include the missing option, specifically:

**Start in text, encapsulate code, no trimming**

This would be effectively "2a without quotes" in which all **_pattern_** whitespace is meaningful. Perhaps no one would go for this option, but we should have it for completeness.

Reply:
Added.